src/Options/Auto.hpp

```diff
@@ -43,6 +43,18 @@ class Auto {
   Auto() = default;
   explicit Auto(T value) : value_(std::move(value)) {}
 
+  // These lines are just to work around a spurious warning.
+  Auto(Auto&&) = default;
+  Auto& operator=(Auto&&) = default;
+#if defined(__GNUC__) and not defined(__clang__) and __GNUC__ >= 12 and __GNUC__ < 13
```

Comment:
Isn't this just `__GNUC__ == 12`? (if so, same for the rest of this PR)

Reply:
It is.  I often write these things as `>= first_broken and < first_optimistically_fixed` (optimistically assuming untested versions have been fixed).  In this case with only one version it's kind of silly, so I'll change it if you like.

Reply:
If you could, yeah so it's obvious it's just this one GCC version. If in the future, we find this is an issue with GCC 13 we can go back to this.